### ISSUE TITLE
ShuffleNet: An Extremely Efficient Convolutional Neural Network for Mobile Devices

### ISSUE THREAD

**Issue by guanfuchen** [guanfuchen/DeepNetModel#10]
related paper

|摘要|
|---|
|We introduce an extremely computation-efficient CNN architecture named ShuffleNet, which is designed specially for mobile devices with very limited computing power (e.g., 10-150 MFLOPs). The new architecture utilizes two new operations, pointwise group convolution and channel shuffle, to greatly reduce computation cost while maintaining accuracy. Experiments on ImageNet classification and MS COCO object detection demonstrate the superior performance of ShuffleNet over other structures, e.g. lower top-1 error (absolute 7.8%) than recent MobileNet [12] on ImageNet classification task, under the computation budget of 40 MFLOPs. On an ARM-based mobile device, ShuffleNet achieves ∼13× actual speedup over AlexNet while maintaining comparable accuracy.|

**Comment by guanfuchen** [comment]
motivation

![image](https://user-images.githubusercontent.com/22321977/50136479-e7fcc180-02d2-11e9-9ebc-6c9780804f44.png)

group convolution

![image](https://user-images.githubusercontent.com/22321977/50136490-ecc17580-02d2-11e9-9bb9-04f83e7ed8e0.png)

channel shuffle operation

![image](https://user-images.githubusercontent.com/22321977/50136507-01057280-02d3-11e9-8acc-34d6c808eec2.png)

channel shuffle and group convolution

![image](https://user-images.githubusercontent.com/22321977/50136521-0ebaf800-02d3-11e9-8ca4-d487a2323afa.png)

![image](https://user-images.githubusercontent.com/22321977/50136532-15496f80-02d3-11e9-8998-8e6aa86d4da4.png)

![image](https://user-images.githubusercontent.com/22321977/50136551-1c707d80-02d3-11e9-9d33-ba927c1a04fc.png)

![image](https://user-images.githubusercontent.com/22321977/50136563-22665e80-02d3-11e9-8013-e3c7de214251.png)

ShuffleNet Unit

![image](https://user-images.githubusercontent.com/22321977/50136573-2d20f380-02d3-11e9-935e-9ae9ce921719.png)

![image](https://user-images.githubusercontent.com/22321977/50136579-3316d480-02d3-11e9-837c-0d94d8d89b42.png)

![image](https://user-images.githubusercontent.com/22321977/50136589-390cb580-02d3-11e9-9446-6ac2372f74d5.png)

network architecture

![image](https://user-images.githubusercontent.com/22321977/50136608-44f87780-02d3-11e9-9d16-0010c65e3726.png)

![image](https://user-images.githubusercontent.com/22321977/50136616-4a55c200-02d3-11e9-8aac-e4e5b93c33b2.png)

**Comment by guanfuchen** [comment]
results

with shuffle vs without shuffle

![image](https://user-images.githubusercontent.com/22321977/50136648-5e99bf00-02d3-11e9-86f8-84c78d62bc7a.png)

group compare

![image](https://user-images.githubusercontent.com/22321977/50136655-6bb6ae00-02d3-11e9-8f21-76700edefef1.png)

![image](https://user-images.githubusercontent.com/22321977/50136664-707b6200-02d3-11e9-9bb3-97e2de3a6202.png)

![image](https://user-images.githubusercontent.com/22321977/50136668-75d8ac80-02d3-11e9-8c30-e5709eff486f.png)

![image](https://user-images.githubusercontent.com/22321977/50136675-7cffba80-02d3-11e9-832f-b28a1f08ed68.png)

![image](https://user-images.githubusercontent.com/22321977/50136681-812bd800-02d3-11e9-9c56-ca95a73803ff.png)

![image](https://user-images.githubusercontent.com/22321977/50136692-85f08c00-02d3-11e9-9136-7d86e68e9ddb.png)

**Comment by guanfuchen** [comment]
conclusions

**Comment by guanfuchen** [comment]
implements

reference
- [shufflenet.py](https://github.com/kuangliu/pytorch-cifar/blob/master/models/shufflenet.py)
- [ShuffleNet.py](https://github.com/ericsun99/ShuffleNet-1g8-Pytorch/blob/master/ShuffleNet.py)，mainly refere this implements
